src/app/parcels/batch/displayComponents/BatchParcelDataGrid.tsx

```diff
@@ -37,65 +45,277 @@ export interface BatchGridDisplayRow {
+                sx={{
+                    "& .MuiDialog-paper": {
+                        padding: "1rem",
+                        borderRadius: "15px",
+                        boxShadow: "0 3px 5px rgba(0,0,0,0.8)",
+                        margin: "auto",
+                        borderColor: submitErrors.length == 0 ? "#a8d49c" : "#990f0f",
+                        borderWidth: "1px",
+                        borderStyle: "solid",
+                    },
+                }}
+            >
+                <DialogTitle
+                    sx={{
+                        fontSize: "1.2rem",
+                        textAlign: "center",
+                    }}
+                >
+                    {dialogMessage}
+                </DialogTitle>
+                {submitErrors.length > 0 && (
+                    <DialogContent sx={{ textAlign: "center" }}>
+                        An unexpected error occurred and some rows did not submit.
+                        <br />
+                        Please try submitting again.
+                    </DialogContent>
+                )}
+                <DialogActions sx={{ justifyContent: "center" }}>
+                    {submitErrors.length > 0 ? (
+                        <>
+                            <Button
+                                variant="contained"
+                                sx={{ margin: "1rem", minWidth: "120px" }}
+                                onClick={() => {
+                                    handleReset();
+                                    setIsSuccessDialogVisible(false);
+                                }}
+                            >
+                                Reset rows and start again
+                            </Button>
+                            <Button
+                                variant="contained"
+                                onClick={() => setIsSuccessDialogVisible(false)}
+                                sx={{ margin: "1rem", minWidth: "120px" }}
+                            >
+                                View unsubmitted rows
+                            </Button>
+                        </>
+                    ) : (
+                        <>
+                            <Button
+                                variant="contained"
+                                sx={{ margin: "1rem", minWidth: "120px" }}
+                                onClick={() => setIsSuccessDialogVisible(false)}
+                            >
+                                Add more rows
+                            </Button>
+                            <Button
+                                variant="contained"
+                                sx={{ margin: "1rem", minWidth: "120px" }}
+                                onClick={handleReturnToParcels}
+                            >
+                                Return to the Parcels Page
+                            </Button>
+                        </>
+                    )}
```

Comment:
I'd ideally like these two options split out into their own components. Something like:
```
                <DialogActions sx={{ justifyContent: "center" }}>
                    {submitErrors.length > 0 ? 
                        <BatchParcelsPostSumbitErrorButtons ... />
                     : <BatchParcelsPostSumbitSuccessButtons ... />
                    )}
```

Reply:
Implemented this. It might be worth doing the same for the separate dialogs for confirmation and post submission.